WalletWasabi.Fluent/Helpers/StartupHelper.cs

```diff
@@ -0,0 +1,63 @@
+using Microsoft.Win32;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Fluent.Helpers
+{
+	public static class StartupHelper
+	{
+		private const string KeyPath = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run";
+
+		public static bool TryModifyStartupSetting(bool runOnSystemStartup)
+		{
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				Assembly assembly = Assembly.GetEntryAssembly() ?? throw new NullReferenceException();
+				string pathToExeFile = assembly.Location[..^4] + ".exe";
+				if (File.Exists(pathToExeFile))
+				{
+					return TryModifyRegistry(runOnSystemStartup, pathToExeFile);
+				}
```

Comment:
There should be `else` too with a logger.LogError statement so that: `logger.LogError($"Path '{pathToExeFile}' does not exist.");

Reply:
right